feat: Refactors featuregate lifecycle management script (#129414)
kubernetes/kubernetes: test/compatibility_lifecycle/README.md

> This directory contains commands for compatibility lifecycle verification.

Reviewer: SGTM, we might want to explain what that is (maybe link to the KEP?)

Author: Updated.

> This directory contains commands for [compatibility lifecycle verification](https://github.com/kubernetes/enhancements/blob/23e8eae624f53771990d2643e20c0937cc54b1f8/keps/sig-architecture/4330-compatibility-versions/README.m). 

Reviewer 2: ```suggestion
This directory contains commands for [compatibility lifecycle verification](https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/4330-compatibility-versions/README.md). 
```

Currently, the following commands are implemented:
```
# Verify feature gate list is up to date

> go run test/featuregate_lifecycle/main.go feature-gates verify

Author: Ah, I missed this one. Thanks.

> go run test/featuregate_lifecycle/main.go feature-gates verify

Reviewer 3: ```suggestion
go run test/compatibility_lifecycle/main.go feature-gates verify
```

# Update feature gate list

> go run test/featuregate_lifecycle/main.go feature-gates update

Reviewer 3: ```suggestion
go run test/compatibility_lifecycle/main.go feature-gates update
```
